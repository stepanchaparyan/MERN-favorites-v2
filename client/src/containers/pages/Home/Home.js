import React from 'react';
import FavItemsList from '../../favItems/FavItemsList/FavItemsList';
import FilterAndSearch from '../../favItems/FilterAndSearch/FilterAndSearch';
import FavItemStat from '../../favItems/FavItemStat/FavItemStat';
import {
  Container,
  Module,
  Info,
  FilterAndSearchContainer,
  FavItemStatContainer
} from './HomeStyled';

const Home = () => {
  return (
    <Container>
      <Module>
        <Info>
          <FilterAndSearchContainer>
            <FilterAndSearch />
          </FilterAndSearchContainer>
          <FavItemStatContainer>
            <FavItemStat />
          </FavItemStatContainer>
        </Info>
        <FavItemsList />
      </Module>
    </Container>
  );
};
export default Home;
