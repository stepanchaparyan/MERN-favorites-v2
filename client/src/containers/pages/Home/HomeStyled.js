import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  justify-content: center;
  color: ${props => props.theme.cadetblue};
  padding: 0 32px 50px 32px;
  font-weight: 600;
  font-size: 24px;
`;

export const Module = styled.div`
  display: flex;
  flex-direction: column;
  width: 1048px;
`;

export const Info = styled.div`
  display: flex;
  justify-content: space-between;
`;

export const FilterAndSearchContainer = styled.div`
  display: flex;
  justify-content: space-between;
  flex-direction: column;
`;

export const FavItemStatContainer = styled.div``;
