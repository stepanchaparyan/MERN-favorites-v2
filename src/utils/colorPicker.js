import theme from '../styles/theme';
import { FAVITEM } from '../constants';

const { red, blue, limeGreen, orange, black } = theme;
const { TYPE } = FAVITEM;

export const colorPicker = category => {
  let color;
  switch (category) {
    case TYPE.FILM:
      color = red;
      break;
    case TYPE.MUSIC:
      color = blue;
      break;
    case TYPE.BOOKS:
      color = limeGreen;
      break;
    case TYPE.OTHER:
      color = orange;
      break;
    default:
      color = black;
  }
  return color;
};
