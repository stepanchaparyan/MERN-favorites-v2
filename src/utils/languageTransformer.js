const dict = {
  // lang - country
  ar: 'sa',
  be: 'bl',
  cs: 'sz',
  da: 'dk',
  de: 'de',
  el: 'gr',
  en: 'us',
  es: 'es',
  fa: 'ir',
  fr: 'fr',
  ja: 'jp',
  hi: 'in',
  hu: 'hu',
  hr: 'hr',
  hy: 'am',
  it: 'it',
  kk: 'kz',
  nl: 'nl',
  no: 'no',
  pl: 'pl',
  pt: 'pt',
  ru: 'ru',
  sk: 'sk',
  sl: 'sl',
  sv: 'se',
  tr: 'tr',
  uk: 'ua'
};

export const languageTransformer = listList => {
  let newList = [];
  listList.map(lang => {
    const item = Object.keys(dict).find(item => item === lang);
    if (item) {
      newList.push(dict[item].toUpperCase());
    }
  });
  return newList;
};

function getKeyByValue(object, value) {
  return Object.keys(object).find(key => object[key] === value);
}

export const setLanguage = (code, setLocale) => {
  const countryCode = Object.values(dict).includes(code.toLowerCase());

  if (countryCode) {
    setLocale(getKeyByValue(dict, code.toLowerCase()));
  }
};
