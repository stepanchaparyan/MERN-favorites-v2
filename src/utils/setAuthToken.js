import axios from 'axios';

const Authorization = 'Authorization';

const setAuthToken = token => {
  if (token) {
    axios.defaults.headers.common = { Authorization: `Bearer ${token}` };
  } else {
    delete axios.defaults.headers.common[Authorization];
  }
};
export default setAuthToken;
