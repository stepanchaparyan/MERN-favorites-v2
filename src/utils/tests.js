import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { shallow } from 'enzyme';

export function renderToJSON(Component, props = {}, Router) {
  if (Router) {
    return renderer
      .create(
        <Router>
          <Component {...props} />
        </Router>
      )
      .toJSON();
  }
  return renderer.create(<Component {...props} />).toJSON();
}

export function renderSnapshot(Component, props = {}) {
  const wrapper = shallow(<Component {...props} />);
  expect(wrapper).toMatchSnapshot();
}
