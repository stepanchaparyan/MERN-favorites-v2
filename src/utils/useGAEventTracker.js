// eslint-disable-next-line no-unused-vars
import React from 'react';
import ReactGA from 'react-ga';

const useGAEventTracker = (category = 'Category') => {
  return (action = 'action', label = 'label') => {
    ReactGA.event({
      category,
      action,
      label,
    });
  };
};

export default useGAEventTracker;
