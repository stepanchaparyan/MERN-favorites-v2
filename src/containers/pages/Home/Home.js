import React, { useEffect } from 'react';
import { useLocation } from 'react-router-dom';
import FavItemsList from '../../favItems/FavItemsList/FavItemsList';
import FilterAndSearch from '../../favItems/FilterAndSearch/FilterAndSearch';
import FavItemStat from '../../favItems/FavItemStat/FavItemStat';
import { Container, Module, Info, FilterAndSearchContainer, FavItemStatContainer } from './HomeStyled';

const Home = () => {
  const { pathname } = useLocation();
  useEffect(() => {
    localStorage.setItem('from', pathname);
  }, []);

  return (
    <Container>
      <Module>
        <Info>
          <FilterAndSearchContainer>
            <FilterAndSearch />
          </FilterAndSearchContainer>
          <FavItemStatContainer>
            <FavItemStat />
          </FavItemStatContainer>
        </Info>
        <FavItemsList />
      </Module>
    </Container>
  );
};
export default Home;
