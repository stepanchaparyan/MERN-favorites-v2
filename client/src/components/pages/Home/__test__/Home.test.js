import { renderSnapshot } from '../../../../utils/tests';
import Home from '../Home.js';

describe('Home component test with Enzyme', () => {
  test('renders correct snapshot', () => {
    renderSnapshot(Home);
  });
});
