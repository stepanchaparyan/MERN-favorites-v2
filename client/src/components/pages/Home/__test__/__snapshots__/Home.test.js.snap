// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component test with Enzyme renders correct snapshot 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <styled.div>
        <FilterAndSearch />
      </styled.div>
      <styled.div>
        <FavItemStat />
      </styled.div>
    </styled.div>
    <FavItemsList />
  </styled.div>
</styled.div>
`;
